data/game_data/main.py

```diff
@@ -0,0 +1,273 @@
+
+import ambulance_game as abg
+
+
+def read_data(path=pathlib.Path("data/parameters/main.csv")):
+    """
+    Read the data contents of the file as a pandas data frame
+    """
+    return pd.read_csv(path)
+
+
+def write_data_to_csv(data, path=pathlib.Path("data/parameters/main.csv")):
+    """
+    Opens `path` in append mode and write the data
+    """
+    with path.open("a", newline="") as outfile:
+        csv_writer = csv.writer(outfile)
+        csv_writer.writerow(data)
+
+
+def initialise_parameters_directory(**problem_parameters):
+    """
+    Creates the parameters directory along with the readme file and the empty
+    main.csv file that will hold all investigated parameters
+    """
+    directory = pathlib.Path("data/parameters/")
+    directory.mkdir(parents=True, exist_ok=True)
+    readme_contents = (
+        "# Parameters"
+        "\n\nThis directory keeps track of all the parameters investigated so far. The"
+        "\ncontents of `main.csv` correspond to parameter values of the model in the"
+        "\nfollowing order:\n"
+    )
+    with (directory / "README.md").open("w") as file:
+        file.write(readme_contents)
+        file.write("".join("\n\t" + key + "," for key, _ in problem_parameters.items()))
+    if not (directory / "main.csv").exists():
+        write_data_to_csv(sorted(problem_parameters.keys()))
+
+
+def generate_data_for_current_parameters(processes, **problem_parameters):
+    """
+    Generates the routing matrix, the payoff matrix for the row player (A) and
+    the payoff matrix for the column player (B), given a set of parameters.
+
+    Returns
+    -------
+    numpy array
+    numpy array
+    numpy array
+    """
+    payoff_matrix_A, payoff_matrix_B, routing_matrix = abg.game.get_payoff_matrices(
+        processes=processes,
+        **problem_parameters,
+    )
+    return routing_matrix, payoff_matrix_A, payoff_matrix_B
+
+
+def write_README_for_current_parameters_directory(readme_path, **problem_parameters):
+    """
+    Writes the readme file (README.md) for the directory of the given set of
+    problem parameters.
+
+    Parameters
+    ----------
+    readme_path : pathlib.Path object
+        the path where the readme file will be located
+    """
+    parameters_string = "".join(
+        "\n\t" + key + " = " + str(value) for key, value in problem_parameters.items()
+    )
+    readme_contents = (
+        "# Experiments for game"
+        "\n\nThis directory consists of the data for the following set of parameters: \n"
+        + "".join(parameters_string)
+        + "\n\nThe directory is structured in the following way:\n\n"
+        "\t|-- main.csv\n"
+        "\t|-- main.npz\n"
+        "\t|-- README.md\n"
+        "\nwhere `main.csv` holds the values of the parameters and "
+        "`main.npz` holds the \nvalues of the generated data.\n\n"
+        "To read the data from `main.npz` in python simply run:\n\n"
+        "```python\n"
+        ">>> import numpy as np\n"
```

Comment:
```suggestion
        "To read the data from `main.npz` in pythonrun:\n\n"
        ""
        "```python\n"
        ">>> import numpy as np\n"
```

Reply:
FYI, it could have been easier to use a multiline string here `"""`.

Reply:
I did at first but it doesn't work well with some features of VScode (will explain better in the meeting).

Reply:
> it doesn't work well with some features of VScode

Sorry I didn't really understand what the problem was during our meeting, but in this case, the features of the editor are the problem not the code.